templates/server-statefulset.yaml

```diff
@@ -110,7 +110,11 @@ spec:
                 fieldRef:
                   fieldPath: metadata.name
             - name: VAULT_CLUSTER_ADDR
+              {{- if .Values.server.ha.clusterAddr }}
```

Comment:
This is correct, but maybe use helm's **default** instead of if/else ? since you're assigning `.Values.server.ha.clusterAddr` to `value` anyway, maybe try, instead of the if/else block:
```
# ...
- name: VAULT_CLUSTER_ADDR
  value: {{ .Values.server.ha.clusterAddr | default (printf "https://$(HOSTNAME).%s-internal:8201" (template "vault.fullname" .)) }}
# ...
```
Just a suggestion, let me know :)

[Documentation reference](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/)

Reply:
Yea, makes sense, I was just trying to keep the syntax consistent with how others were doing things,

Reply:
though, not sure why you have the`(template "vault.fullname" .)` in there.. typo?

Reply:
oh, you're replacing with template fullname.. that isn't required as $(HOSTNAME) is an environment variable setting on the host itself.. it resolves correctly as a variable.

Reply:
Forget it, I know what you're doing. :)

Reply:
The default function won't parse the "template" function, so this won't work unfortunately.

Reply:
My bad, in that snippet I just followed the documentation blindly, as you stated it might not work then. Helm charts can be a pain.

Reply:
All good, when you said it, I figured i'd try, didn't work though.. made me sad :(